src/Components/Entry/Entry.tsx

```diff
@@ -32,6 +46,30 @@ export const Entry: FC = () => {
     const handleFrequencyChange = (event: SelectChangeEvent<string> ) => {
         setFrequency(event.target.value)
     }
+     
+
+
+  
+
+    const handleSubmit = async (event: BaseSyntheticEvent) => {
+        event.preventDefault();
+        //how do i set this req.body as entryReqBody?
```

Comment:
My _guess_ would be that we would define the interface and then say that the request object is an instance of the interface:

interface entryReqBody {
            accountId:string
            entryType:string,
            date:string?date? not sure here
            creditDebit:string
            amount:Number?Float?
            title:string,
            note:string,
            frequency:string
};

const body: entryReqBody = {
            accountId,
            entryType,
            date,
            creditDebit,
            amount,
            title,
            note,
            frequency
};

Try it! If it doesn't work it's YiFan's fault!

Reply:
Seems reasonable to me, but I'm not sure why this needs to be done? The types of each segment of the data should be passed through from their initial set? On the backend, the Request type will allow for access to the data then you can break the data out for input into a new Entry.